src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinAnnotationIntrospector.kt

```diff
@@ -92,12 +89,18 @@ internal class KotlinAnnotationIntrospector(private val context: Module.SetupCon
     private fun AnnotatedParameter.hasRequiredMarker(): Boolean? {
         val member = this.member
         return when (member) {
-            is Constructor<*> -> member.kotlinFunction?.isParameterRequired(index)
-            is Method         -> member.kotlinFunction?.isParameterRequired(index)
+            is Constructor<*> -> member.kotlinFunction?.isConstructorParameterRequired(index)
+            is Method         -> member.kotlinFunction?.isMethodParameterRequired(index)
             else              -> null
         }
     }
 
+    private fun KFunction<*>.isConstructorParameterRequired(index: Int): Boolean =
+            this.isParameterRequired(index)
+
+    private fun KFunction<*>.isMethodParameterRequired(index: Int): Boolean =
+            this.isParameterRequired(index + 1) // Skip the companion object passed as the first param
```

Comment:
>  If this callable requires a this instance or an extension receiver parameter, they come first in the list in that order.
(http://kotlinlang.org/api/latest/jvm/stdlib/kotlin.reflect/-k-callable/parameters.html)
Is this true for all methods that can be annotated? Filtering out the first param that is of `kind = INSTANCE` probably also works.

Reply:
I've played with it a bit and, thanks your comment, simplified and unified the code. Thanks @mattibickel